### PR TITLE
[DOC] Updates query syntax

### PR DESCRIPTION
**Pull request by nermiller**
Updates wildcard matching, corrects w-pattern usage, adds DIALECT 3 information, plus copyedits

### CODE REVIEW

File: docs/docs/reference/Query_Syntax.md [hunk 1]
```diff
@@ -3,130 +3,139 @@ title: "Query syntax"
...
-* Multi-word phrases simply a list of tokens, e.g. `foo bar baz`, and imply intersection (AND) of the terms.
-* Exact phrases are wrapped in quotes, e.g `"hello world"`.
-* OR Unions (i.e `word1 OR word2`), are expressed with a pipe (`|`), e.g. `hello|hallo|shalom|hola`.
-* NOT negation (i.e. `word1 NOT word2`) of expressions or sub-queries. e.g. `hello -world`. As of version 0.19.3, purely negative queries (i.e. `-foo` or `-@title:(foo|bar)`) are supported.
-* Prefix/Infix/Suffix matches (all terms starting/containing/ending with a term) are expressed with a `*`. For performance reasons, a minimum term length is enforced (2 by default, but is configurable).
-* Wildcard pattern matches: `w'foo*bar?'`.
-* A special "wildcard query" that returns all results in the index - `*` (cannot be combined with anything else).
+* Multiword phrases are lists of tokens, for example, `foo bar baz`, and imply intersection (AND) of the terms.
+* Exact phrases are wrapped in quotes, for example, `"hello world"`.
+* `OR` unions are expressed with a pipe (`|`), for example, `hello|hallo|shalom|hola`.
```
Comment (oshadmi) [inline docs/docs/reference/Query_Syntax.md:13]: There are related fixes in DIALECT 2
See "Examples of queries impacted" in [v2.4.3](https://github.com/RediSearch/RediSearch/releases/tag/v2.4.3/) Release Notes

> Consider a simple query with negation -hello world.
> ...

> Another example illustrating the differences in parser behavior
> hello world | "goodbye" moon.
> 
> In DIALECT 1, this query is interpreted as searching for (hello world | "goodbye") moon.
> In DIALECT 2, this query is interpreted as searching for either hello world OR "goodbye"  moon
> 
>

**Review by oshadmi — COMMENTED** [review]
Looks good!
Few comments

File: docs/docs/reference/Query_Syntax.md [hunk 2]
```diff
@@ -10,20 +10,35 @@ You can use simple syntax for complex queries using these rules:
...
 * Multiword phrases are lists of tokens, for example, `foo bar baz`, and imply intersection (AND) of the terms.
 * Exact phrases are wrapped in quotes, for example, `"hello world"`.
-* `OR` unions are expressed with a pipe (`|`), for example, `hello|hallo|shalom|hola`.
-* `NOT` negation of expressions or subqueries is expressed with a subtraction symbol (`-`), for example, `hello -world`. Purely negative queries (for example, `-foo` or `-@title:(foo|bar)`) are supported **(as of v0.19.3)**.
+* `OR` unions are expressed with a pipe (`|`), for example, `hello|hallo|shalom|hola`. 
+
+  {{% alert title="Note" color="warning" %}}
+  Consider the differences in parser behavior in example `hello world | "goodbye" moon`:
+  * In DIALECT 1, this query is interpreted as searching for `(hello world | "goodbye") moon`.
+  * In DIALECT 2, this query is interpreted as searching for either `hello world` **OR** `"goodbye" moon`.
```
Comment (oshadmi) [inline docs/docs/reference/Query_Syntax.md:18]: Also in other occurrences
```suggestion
  * In DIALECT 2 or greater, this query is interpreted as searching for either `hello world` **OR** `"goodbye" moon`.
```

File: docs/docs/reference/Query_Syntax.md [hunk 3]
```diff
@@ -3,130 +3,164 @@ title: "Query syntax"
...
+  * In DIALECT 2 or greater, this query is interpreted as searching for either `hello world` **OR** `"goodbye" moon`.
+  {{% /alert %}}
+  
+* `NOT` negation of expressions or subqueries is expressed with a subtraction symbol (`-`), for example, `hello -world`. Purely negative queries (for example, `-foo` or `-@title:(foo|bar)`) are also supported.
+
+  {{% alert title="Note" color="warning" %}}
+  Consider a simple query with negation `-hello world`:
+  * In DIALECT 1, this query is interpreted as "find values in any field that does not contain `hello` **AND** does not contain `world`". The equivalent is `-(hello world)` or `-hello -world`.
+  * In DIALECT 2, this query is interpreted `as -hello` **AND** `world` (only `hello` is negated).
+  * In DIALECT 2, You could achieve the default behavior as in dialect 1 by updating your query to `-(hello world)`.
```
Comment (nermiller) [inline docs/docs/reference/Query_Syntax.md:27]:
```suggestion
  * In DIALECT 2, to achieve the default behavior of dialect 1, update your query to `-(hello world)`.
```

File: docs/docs/reference/Query_Syntax.md [hunk 4]
```diff
@@ -57,13 +57,13 @@ You can specify field modifiers in a query, and not just by using the `INFIELDS`
...
+If a field modifier precedes multiple words or expressions, it applies only to the adjacent expression with DIALECT 1. With DIALECT 2 or greater, you extend the query to other fields.
 
 Consider this simple query: `@name:James Brown`. Here, the field modifier `@name` is followed by two words: `James` and `Brown`.
 
 * In DIALECT 1, this query would be interpreted as "find `James Brown` in the `@name` field".
-* In DIALECT 2, this query will be interpreted as "find `James` in the `@name` field **AND** `Brown` in **ANY** text field. In other words, it would be interpreted as `(@name:James) Brown`.
-* In DIALECT 2, you can achieve the default behavior as in DIALECT 1 by updating your query to `@name:(James Brown)`.
+* In DIALECT 2 or greater, this query will be interpreted as "find `James` in the `@name` field **AND** `Brown` in **ANY** text field. In other words, it would be interpreted as `(@name:James) Brown`.
+* In DIALECT 2 or greater, you can achieve the default behavior as in DIALECT 1 by updating your query to `@name:(James Brown)`.
```
Comment (nermiller) [inline docs/docs/reference/Query_Syntax.md:66]:
```suggestion
* In DIALECT 2 or greater, to achieve the default behavior of DIALECT 1, update your query to `@name:(James Brown)`.
```

File: docs/docs/reference/Query_Syntax.md [hunk 5]
```diff
@@ -23,8 +23,8 @@ You can use simple syntax for complex queries using these rules:
   {{% alert title="Note" color="warning" %}}
   Consider a simple query with negation `-hello world`:
   * In DIALECT 1, this query is interpreted as "find values in any field that does not contain `hello` **AND** does not contain `world`". The equivalent is `-(hello world)` or `-hello -world`.
-  * In DIALECT 2, this query is interpreted `as -hello` **AND** `world` (only `hello` is negated).
-  * In DIALECT 2, to achieve the default behavior of dialect 1, update your query to `-(hello world)`.
+  * In DIALECT 2 or greater, this query is interpreted `as -hello` **AND** `world` (only `hello` is negated).
+  * In DIALECT 2 or greater, to achieve the default behavior of dialect 1, update your query to `-(hello world)`.
```
Comment (nermiller) [inline docs/docs/reference/Query_Syntax.md:27]:
```suggestion
  * In DIALECT 2 or greater, to achieve the default behavior of DIALECT 1, update your query to `-(hello world)`.
```

**Review by oshadmi — APPROVED** [review]
👍🏼 LGTM

**Review by oshadmi — APPROVED** [review]
👌🏼

**Comment by nermiller** [comment]
@rustyedges oh wow, i'll fix that asap. thanks for bringing it up!

**Comment by nermiller** [comment]
@rustyedges this issue has been fixed!

**Pull request merged**